Islandora/islandora
# Mjordan issue 1521 redo

**Github issues**: https://github.com/Islandora/documentation/issues/1549 and https://github.com/Islandora/documentation/issues/1550.

* Supersedes https://github.com/Islandora/islandora/pull/827

# What does this Pull Request do?

Accounts for the fact that @mjordan and @dannylamb are idiots. Specifically, this is a redo of the PRs https://github.com/Islandora/islandora/pull/778 and https://github.com/Islandora/islandora/pull/782, which Mark incorrectly opened against and were merged against the master branch.  Danny erroneously merged them :tada:

# What's new?

This PR improves https://github.com/Islandora/islandora/pull/827 by replacing calls to `Drupal::service` with dependency injection.

* Same things that were new in PRs https://github.com/Islandora/islandora/pull/778 and https://github.com/Islandora/islandora/pull/782:
    * Adds a new controller at `admin/reports/islandora/rdf_mappings` that generates and renders the report (including updating the menu links and routing YAML files as required)
    * Updates the `getUriFieldNamesForTerms()` function in the Islandora Utilities service

# How should this be tested?

1. Check out this branch. Within the `islandora` module directory:
   1. `git fetch origin`
   1. `git checkout mjordan-issue-1521-redo`
1. Clear your cache (`drush cr`)
1. Visit Admin > Reports > Field and term RDF mappings (refresh if you don't see this new item)
1. You should see the report listing fields and their RDF property mappings, and near the bottom, all taxonomies with terms that have external URIs / authority links. It's a long list.

# Additional Notes:

* Does this change the interface, add a new feature, or otherwise change behaviours that would require updating documentation? Yes, but only adds more information to an existing report.
* Does this change add any new dependencies? No.
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? No.
* Could this change impact execution of existing code? Yes.

# Interested parties
@Islandora/8-x-committers
